Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4483,104 +4473,38 @@ namespace ScriptCanvasEditor
         auto result = ScriptEvents::Editor::UpdateMenuItemsEnabled(m_activeGraph);
         ui->actionAdd_Script_Event_Helpers->setEnabled(result.m_addHelpers);
         ui->actionClear_Script_Event_Status->setEnabled(result.m_clear);
-        ui->actionParse_As_Script_Event->setEnabled(result.m_parse);
-        ui->actionSave_As_ScriptEvent->setEnabled(result.m_save);
+        ui->actionSave_ScriptEvent->setEnabled(result.m_save);
+        ui->actionSave_As_ScriptEvent->setEnabled(result.m_saveAs);
     }
 
-    void MainWindow::OnScriptEventOpen()
+    void MainWindow::OnScriptEventOpen(const AZ::IO::Path& filePath)
     {
-        AZStd::pair<ScriptCanvas::SourceHandle, AZStd::string> result = ScriptEvents::Editor::OpenAction();
+        ScriptCanvas::SourceHandle sourceAsset(ScriptEvents::Editor::OpenAction(filePath));
 
-        if (result.first.Get())
-        {
-            OpenScriptCanvasAssetImplementation(result.first, Tracker::ScriptCanvasFileState::UNMODIFIED);
-        }
-        else
+        if (sourceAsset.IsGraphValid())
         {
-            if (!result.second.empty())
-            {
-                QMessageBox mb
-                    ( QMessageBox::Warning
-                    , tr("Failed to open ScriptEvent file into ScriptCanvas Editor.")
-                    , result.second.c_str()
-                    , QMessageBox::Close
-                    , nullptr);
-
-                mb.exec();
-            }
+            AddRecentFile(sourceAsset.AbsolutePath().c_str());
+            OpenScriptCanvasAssetImplementation(sourceAsset, Tracker::ScriptCanvasFileState::UNMODIFIED);
         }
     }
 
-    void MainWindow::OnScriptEventParseAs()
+    bool MainWindow::OnScriptEventSave(Save save)
     {
-        if (!m_activeGraph.IsGraphValid())
-        {
-            return;
-        }
-
-        AZStd::pair<bool, AZStd::vector<AZStd::string>> result = ScriptEvents::Editor::ParseAsAction(m_activeGraph);
+        bool saveInPlace = save == Save::InPlace ? true : false;
```

Comment:
Nit: Can simplify this
```suggestion
        bool saveInPlace = (save == Save::InPlace);
```

Reply:
done